### PR TITLE
Update CMake configuration of new libraries

### PR DESCRIPTION
**Pull request by aggarw13**
The FreeRTOS Device Software console requires that the CMake metadata provide header files path, so that they are included in the downloaded ZIP. Thus, update the `core_mqtt`, `logging`, `retry_utils` and `transport_interface_secure_sockets` CMake targets to include header files in their **source** list

### CODE REVIEW

File: libraries/abstractions/retry_utils/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,18 @@
 afr_module(name retry_utils)
 
+# Metadata for the module.
+afr_set_lib_metadata(ID "retry_utils")
+afr_set_lib_metadata(DESCRIPTION "This library represents the exponential backoff with jitter retruy utility.")
```
Comment (aggarg) [inline libraries/abstractions/retry_utils/CMakeLists.txt:5]: retry

File: libraries/abstractions/retry_utils/CMakeLists.txt [hunk 2]
```diff
@@ -1,9 +1,18 @@
 afr_module(name retry_utils)
```
Comment (aggarg) [inline libraries/abstractions/retry_utils/CMakeLists.txt:1]: Capitalize NAME here too?

File: libraries/core_mqtt.cmake [hunk 3]
```diff
@@ -11,9 +11,21 @@ afr_set_lib_metadata(IS_VISIBLE "true")
...
+    file(GLOB mqtt_public_include_header_files
+              LIST_DIRECTORIES false
+              ${mqtt_public_include_dir}/* )
+    list(APPEND MQTT_HEADER_FILES ${mqtt_public_include_header_files})
+endforeach()
+
 afr_module_sources(
     ${AFR_CURRENT_MODULE}
     PRIVATE
+        ${MQTT_HEADER_FILES}
```
Comment (leegeth) [inline libraries/core_mqtt.cmake:28]: Can't we directly use ${MQTT_INCLUDE_PUBLIC_DIRS} in here?

Comment (aggarw13) [reply libraries/core_mqtt.cmake:28]: We need to have the entire file path like `coreMQTT/source/include/core_mqtt.h` to add the header file in the metadata. The `{MQTT_INCLUDE_PUBLIC_DIRS}` only contains the directory path to the header files.

**Review by aggarg — APPROVED** [review]

File: libraries/core_mqtt.cmake (hunk 3 above)
Comment (leegeth) [reply libraries/core_mqtt.cmake:28]: Ok. Thanks.

**Review by leegeth — DISMISSED** [review]

**Review by leegeth — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Review by leegeth — APPROVED** [review]

**Pull request merged**